### DISCUSSION TITLE
Genuine question: Given the webserver can read in the metadatabase could we store tls certificates inside the metadatabase ?

### DISCUSSION THREAD

**Discussion by ginolegigot** [apache/airflow#34371] [Q&A]
Hello!
It would be great if we could store the certificate inside the metadatabase as a connection maybe so that we dont need to expose these TLS certificates on the VM filesystem. So i'm asking if this is possible or even a decent idea ?
(in case of it is not clear, I'm talking about TLS certificates for an https connection to the webserver)

**Comment by potiuk** [comment]
No. Not good idea. The **righ** way of doing it is to shield the webserver by a proxy to terminate TLS. This should be part of the deployment and the only way to make it securely. It does not add any value if it's kept in the database, because someone who has access to the VM, might very well connect to the database and read it from there.
